aptos-move/framework/aptos-stdlib/sources/math64.move

```diff
@@ -20,6 +23,16 @@ module aptos_std::math64 {
         }
     }
 
+    /// Returns a * b / c going through u128 to prevent intermediate overflow
+    public inline fun mul_div(a: u64, b: u64, c: u64): u64 {
```

Comment:
This redundancy is so unfortunate :( I can't think of any way to reduce this though - may be interface can help but that's much further down the road. Do you have any ideas here?

Reply:
Yes I dont think there is a solution here right now. We do need typeclasses